Source/Core/DolphinQt2/DPI.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/DPI.h"
+
+#include <algorithm>
+
+#include <QGuiApplication>
+#include <QPaintDevice>
+#include <QPixmap>
+#include <QScreen>
+#include <QSize>
+#include <QString>
+
+namespace DPI
+{
+double GetScale()
+{
+  return static_cast<QGuiApplication*>(QGuiApplication::instance())->devicePixelRatio() * 1.1;
```

Comment:
Have you tried this on a High DPI system? `devicePixelRatio()` will return the exact scale (1x, 1.5x 2x) without needing anymore calculations.

Also, is there a reason why you went with `static_cast<QGuiApplication*>(QGuiApplication::instance())->devicePixelRatio()` instead of `QGuiApplication::primaryScreen()->devicePixelRatio()` (maybe that's your "too small" issue)?